operator/pkg/controllers/controller.go

```diff
@@ -106,3 +119,34 @@ func (c *GenericController) reconcile(ctx context.Context, resource Object, pers
 	}
 	return *result, nil
 }
+
+func (c *GenericController) finalizeSubResources(ctx context.Context, resource Object, finalizers []string) (*reconcile.Result, error) {
+	result, err := c.Controller.Finalize(ctx, resource)
+	if err != nil {
+		return result, fmt.Errorf("finalizing resource controller %v, %w", c.Controller.Name(), err)
+	}
+	// Remove finalizer for this controller
+	resource.SetFinalizers(finalizers)
+	zap.S().Infof("[%s] Successfully deleted resources", resource.GetName())
+	return result, nil
+}
+
+func ttlExpiredForControlPlane(resource Object) bool {
+	cp, ok := resource.(*v1alpha1.ControlPlane)
+	if !ok {
+		return false
+	}
+	if cp.Spec.TTL != "" {
+		duration, err := time.ParseDuration(cp.Spec.TTL)
+		if err != nil {
+			zap.S().Errorf("parsing TTL duration, %w", err)
+			return false
+		}
+		deleteAfter := resource.GetCreationTimestamp().Add(duration)
```

Comment:
minor: i think this logic can be reused to check ttls for other components too..like data plane. However not a blocker.